File: android/sample-ui/src/main/kotlin/com/usesmileid/sampleapps/ui/components/UseSmileIDSampleScanGlyph.kt
package com.usesmileid.sampleapps.ui.components

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.usesmileid.sampleapps.ui.R
import com.usesmileid.sampleapps.ui.theme.UseSmileIDSampleTheme

/** The scan target: one exported asset rather than a drawn reticle, so its proportions cannot drift. */
@Composable
fun UseSmileIDSampleScanGlyph(
    modifier: Modifier = Modifier,
    size: Dp = SCAN_GLYPH_SIZE,
    /** The reticle carries the scanner's state over a live camera, so its tint is the caller's. */
    tint: Color = UseSmileIDSampleTheme.colors.textTitle,
) = UseSmileIDSampleIcon(
    id = R.drawable.sample_ic_scan_glyph,
    tint = tint,
    modifier = modifier,
    size = size,
)

private val SCAN_GLYPH_SIZE = 279.dp

File: flutter/sample_ui/lib/src/components/use_smileid_sample_scan_glyph.dart
import 'package:flutter/material.dart';

import '../theme/use_smileid_sample_theme.dart';
import '../tokens/smile_icons.dart';
import 'use_smileid_sample_icon.dart';

/// The scan target: one exported asset rather than a drawn reticle, so its proportions are the
/// design's own. The tint is a parameter because the reticle carries scanner state over a camera.
class UseSmileIDSampleScanGlyph extends StatelessWidget {
  /// [size] is the design's 279, not a scale token — no token reaches it.
  const UseSmileIDSampleScanGlyph({
    this.size = _scanGlyphSize,
    this.tint,
    super.key,
  });

  /// The square the glyph is drawn into.
  final double size;

  /// The bracket colour; the title colour where there is no camera behind it.
  final Color? tint;

  @override
  Widget build(BuildContext context) => UseSmileIDSampleIcon(
    asset: SmileIcons.scanGlyph,
    tint: tint ?? UseSmileIDSampleTheme.colorsOf(context).textTitle,
    size: size,
  );
}

/// The design's own 279; no scale token reaches it.
const double _scanGlyphSize = 279;
